releasenotes/notes/bugfix-union-7da79789134a3028.yaml

```diff
@@ -0,0 +1,24 @@
+---
+fixes:
+  - |
+    Previously, :func:`~retworkx.digraph_union` would falsely keep or delete edges
+    if argument `merge_edges` is set to true. This has been fixed and an edge from
+    the second graph will be skipped if both its endpoints were merged to nodes from
+    the first graph and these nodes already share an edge with equal weight data.
+    Fixed `#432 <https://github.com/Qiskit/retworkx/issues/432>`__
+features:
```

Comment:
I wonder if it's worth noting that `merge_nodes` and `merge_edges` for `digraph_union` are optional now and default False because previously they were both required I think.

Reply:
Done in 088f27d